### ISSUE TITLE
PNP and basedir trailing slashes

### ISSUE THREAD

**Issue by ef4** [stefanpenner/resolve-package-path#24]
When calling PNP's `resolveToUnqualified` method, when the issuer represents a directory the issuer must have a trailing slash.

As far as I can tell, resolve-package-path deals only in base directories, never base *files*, so I think it should always append a trailing slash to basedir before calling `resolveToUnqualified`.

Here are reproduction steps:

1. Clone this repo.
2. cd `resolve-package-path`
3. `volta pin node@13` 
4. `volta pin yarn@1.22.4`
5. `yarn set version berry`
6. `yarn`
7. Start interactive node with `yarn node`
8. Evaluate these steps:

    ```js
    // load our requirePackagePath function
    rpp = require('.')

    // use it to locate chai. This one happens to work even without a trailing slash, I think 
    // because the issuer is a real path on the filesystem
    chaiDir = path.dirname(rpp('chai', '.'))
    // Result: '/Users/edward/hacking/resolve-package-path/.yarn/cache/chai-npm-4.2.0-b777309e5a-2.zip/node_modules/chai'

    // this is a valid dependency of chai, but it returns null! In this case, the issuer is a hypothetical
    // directory inside a zip file that only yarn pnp understands. I suspect that is why the
    // trailing slash is mandatory.
    rpp('deep-eql', chaiDir)
    // Result: null

    // you can make it get the right answer like this:
    rpp('deep-eql', chaiDir + '/')
    // Result: '/Users/edward/hacking/resolve-package-path/.yarn/cache/deep-eql-npm-3.0.1-9a66c09c65-2.zip/node_modules/deep-eql/package.json'
    ```

**Comment by stefanpenner** [comment]
@ef4 do you see any downsides or risks?

Is it actually as easy as something like?
```js
const path = require('path');

 if (dir.charAt(dir.length - 1) !== path.sep) {
  dir = `${dir}${path.sep}`;
 }
```

If so, I'll gladly make this change.

**Comment by ef4** [comment]
Yes, I think that's all that's required.

**Comment by stefanpenner** [comment]
@ef4 I have not forgotten this, I have just been out due to some health stuff.

**Comment by merceyz** [comment]
Might not be needed anymore after https://github.com/yarnpkg/berry/pull/2353 got merged and released in 3.0.0-rc.1

**Comment by stefanpenner** [comment]
PR https://github.com/stefanpenner/resolve-package-path/pull/46/ (should fix pnp usage prior to 3.0.0-rc.1 etc)
